neqo-transport/src/connection.rs

```diff
@@ -3301,58 +3450,126 @@ mod tests {
-        assert_eq!(lr_module.get_timer(), 348_125_000);
+        // Timer expires, packet 1 is lost.
+        lr_module.on_loss_detection_timeout(start + Duration::from_micros(120_390));
+        assert_values(
+            &lr_module,
+            Duration::from_micros(85_000),
+            Duration::from_micros(98_125),
+            Duration::from_micros(41_250),
+            Duration::from_micros(85_000),
+            [None, None, None],
+        );
+        assert_eq!(
+            lr_module.get_timer(start + Duration::from_nanos(120_390_000)),
+            Some(Duration::from_nanos(348_125_000) - Duration::from_nanos(120_390_000))
+        );
 
         // dupacks loss detection. ackes 0, 2 and 6, markes packet 3 as lost.
         lr_module.on_ack_received(
             PNSpace::ApplicationData,
             6,
             vec![(0, 0), (2, 2), (6, 6)],
-            2000,
-            130_000_000,
+            Duration::from_micros(2000),
+            start + Duration::from_nanos(130_000_000),
+        );
+        assert_values(
+            &lr_module,
+            Duration::from_micros(70_000),
+            Duration::from_micros(94_609),
+            Duration::from_micros(37_968),
+            Duration::from_micros(70_000),
+            [None, None, Some(start + Duration::from_micros(146_435))],
+        );
+        assert_eq!(
+            lr_module.get_timer(start + Duration::from_nanos(130_000_000)),
+            Some(Duration::from_nanos(146_435_000) - Duration::from_nanos(130_000_000))
         );
-        assert_values(&lr_module, 70_000, 94_609, 37_968, 70_000, [0, 0, 146_435]);
-        assert_eq!(lr_module.get_timer(), 146_435_000);
 
         // Timer expires, packet 4 is lost.
-        lr_module.on_loss_detection_timeout(146_500_000);
-        assert_values(&lr_module, 70_000, 94_609, 37_968, 70_000, [0, 0, 156_435]);
-        assert_eq!(lr_module.get_timer(), 156_435_000);
+        lr_module.on_loss_detection_timeout(start + Duration::from_nanos(146_500_000));
+        assert_values(
+            &lr_module,
+            Duration::from_micros(70_000),
+            Duration::from_micros(94_609),
+            Duration::from_micros(37_968),
+            Duration::from_micros(70_000),
+            [None, None, Some(start + Duration::from_micros(156_435))],
+        );
+        assert_eq!(
+            lr_module.get_timer(start + Duration::from_nanos(146_500_000)),
+            Some(Duration::from_nanos(156_435_000) - Duration::from_nanos(146_500_000))
+        );
 
         // Timer expires, packet 5 is lost.
-        lr_module.on_loss_detection_timeout(156_500_000);
-        assert_values(&lr_module, 70_000, 94_609, 37_968, 70_000, [0, 0, 0]);
+        lr_module.on_loss_detection_timeout(start + Duration::from_nanos(156_500_000));
+        assert_values(
+            &lr_module,
+            Duration::from_micros(70_000),
+            Duration::from_micros(94_609),
```

Comment:
This test needs some constants.  Badly.

Reply:
How could we do this best? Or, what would we call them?

`const TEST_LOSS_RECOVERY2_LATEST_RTT_AFTER_PKT_3_LOST:Duration = Duration::from_micros(70_000)`?

Reply:
Probably something simple like STEP1, STEP2, etc...  Where we need to provide the difference, we can use `STEP2 - STEP1`.